clients/drcachesim/tracer/physaddr.h

```diff
@@ -52,10 +54,11 @@ class physaddr_t {
     virtual2physical(addr_t virt);
 
 private:
-    // Assumed to be single-threaded
 #ifdef LINUX
     addr_t last_vpage_;
     addr_t last_ppage_;
+    // XXX: An app with thousands of threads might hit open file limits.
```

Comment:
Does this use one of DR's private FDs? I don't think so, as the implementation doesn't use `dr_file_open`. It probably should right? But then I'd be concerned about exhausting the small number of available private FDs (96). A lock would cause too much contention I suppose, so that's not the solution...

Reply:
It does use a private FD: "open" is redirected to "os_open_protected".  So a many-threaded app will indeed use that up.  Hmm.

Reply:
> "open" is redirected to "os_open_protected"

I thought only the app's `open`s are redirected.

Reply:
"Redirected" means pointing to a different function, done by the private loader to client and private libraries: https://github.com/DynamoRIO/dynamorio/blob/master/core/unix/loader.c#L1438

For the app using SYS_open: I don't think DR does anything there.  Looking: just some debug stuff and some emulation stuff.

Reply:
> For the app using SYS_open

Oh right, got it.

> "Redirected" means pointing to a different function, done by the private loader to client and private libraries:

Okay.

Reply:
I think for now I will augment the comment to talk about using up the private FDs but not try to solve it yet; the goal here is to get simple apps/benchmarks working.  We'll list scaling up the thousands of threads as a separate issue to solve.

Reply:
Expanded the comment pointing to #4014.